opslevel/common.go

```diff
@@ -37,6 +38,31 @@ import (
 
 const providerIssueUrl = "https://github.com/OpsLevel/terraform-provider-opslevel/issues"
 
+type CommonResourceClient struct {
+	client *opslevel.Client
+}
+
+// Configure sets up the OpsLevel client for datasources and resources
+func (d *CommonResourceClient) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*opslevel.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
```

Comment:
suggestion: make this a helper

Reply:
This is non-trivial since `resp` and `req` in this if statement are two different types from two different packages. Passing on this suggestion as per offline conversation